### ISSUE TITLE
Hide tabs headers when setting the app to fullscreen.

### ISSUE THREAD

**Issue by Legendarior** [warpdotdev/Warp#1559]
### Describe the solution you'd like?

```text
Hide tabs header bar when the application is set to fullscreen (as is fullscreen in new desktop space, not fill screen).
Especially useful when working with fullscreen TUI applications like Neovim, Emacs or Helix.
You could implement a solution similar to iTerm2, where when you hover the cursor at the top of the screen it shows the tabs, but else hide them.
```

### Is your feature request related to a problem? Please describe.

```text
It's mostly a QoL issue to regain screen real-estate when working with fullscreen TUI applications.
```

**Comment by Brabalawuka** [comment]
Please hide this blank space~

![image](https://user-images.githubusercontent.com/31028570/176392346-078982d1-3cf6-4570-b66d-4216ae4a7a66.png)

**Comment by elviskahoro** [comment]
Thanks for submitting. I think there are issues already open for this. Will link them when I find them later

**Comment by joehannes** [comment]
hey, apart from just wanting to report a very similar thingy and totally agreeing I want to propose an idea:

1. First, hiding the tabs in fullscreen mode would is the missing feature that's hindering me atm to switch to warp

An Illustration:

In Alacritty:
![image](https://user-images.githubusercontent.com/63741107/180664502-4e475282-bc63-4d4b-bfe9-326753caa2bc.png)

In Warp:
![image](https://user-images.githubusercontent.com/63741107/180664522-f3421c41-4c5b-401b-b9e9-6e7a2991004d.png)

That thick tabline of is practically of no use to me if using tmux and therefor ... ufff ...

2. Please
* Enable hiding the tabline altogether if one tab present only
* Enable hiding it altogether in fullscreen mode eg. as iterm2 does

**A proposition:**
Since you got feature like remember last layout and such, at least as soon as you got profiles for the layouts/tabs as well ...
* wouldn't it be nice to pimp up that tabline?
* to make it an actual replacement for tmux?
* maybe even parsing and incorporating .tmuxrc - being able to read and represent .tmux-layouts would be awesome!!???

Anyway, wether you integrate tmux into your warp/tabs or create your own schema, tabline-components and tab-adjustable-look-and-feel would be awesome!!
* basically to be something like a tmux statusline
* having slanted/fancy tabs
* being able to add arbitrary/custom (cmds as) components like "IP address" or CWD or GIT Repo-Info

Having this, one would be able to ditch tmux altogether for warp-profiles, or at least switch to warp once this huge out of my tmux/nvim-colors tabline is gone in my fullscreen, thx :-) !!!

**Comment by warpdotdev-dx** [comment]
Can you elaborate on these two GitHub discussions as well:
GitHub discussion on Session management and profiles: <https://github.com/warpdotdev/Warp/discussions/697>
GitHub issue for tmux support: https://github.com/warpdotdev/Warp/discussions/501

**Comment by joehannes** [comment]
> Can you elaborate on these two GitHub discussions as well: GitHub discussion on Session management and profiles: <#697> GitHub issue for tmux support: #501

https://github.com/warpdotdev/Warp/discussions/501#discussioncomment-3257131

**Comment by JulienLecoq** [comment]
> ### Describe the solution you'd like?
> ```
> Hide tabs header bar when the application is set to fullscreen (as is fullscreen in new desktop space, not fill screen).
> Especially useful when working with fullscreen TUI applications like Neovim, Emacs or Helix.
> You could implement a solution similar to iTerm2, where when you hover the cursor at the top of the screen it shows the tabs, but else hide them.
> ```
> 
> ### Is your feature request related to a problem? Please describe.
> ```
> It's mostly a QoL issue to regain screen real-estate when working with fullscreen TUI applications.
> ```
> 
> ### Additional context
> _No response_
> 
> ### Warp Internal (ignore) - linear-label:efcea52d-0ab4-4a92-a41b-93c3e244f0a0
> _No response_

I would love that too, this is also what prevent me from using Warp (I hate tabs and also as others mentioned the blank space when using app such as Neovim is a bit annoying).

**Comment by WillJRoper** [comment]
I second this. I really want to fully switch to Warp, but without the control to make the terminal as minimal as possible, I can't leave behind iterm2 for my Emacs usage. The wasted real estate for tabs that aren't used when switching between buffers is a roadblock to full adoption, which I'm sure is true for many using emacs and other terminal-bound editors.

#697 (profiles) is also essential for this to have a specific profile for it and enable start-up commands to automatically connect to the daemon and run the client when opening an "Emacs" profile. 

Less of a roadblock but more of a "nice to have" would be control over **where** the tab bar is which iterm2 gives you. It may be unorthodox but I like to have my tabs at the bottom. Additionally, the text indicator of the shortcut for each tab is nice for quick switching at a glance when lots of tabs are open. 

A suggestion for a nice approach: The way Arc treats browser tabs with a sidebar that can be hidden or shown with a keyboard shortcut would be incredible for maintaining functionality while enabling minimalism. If something of this ilk could be implemented I'd be able to fully convert (with the hopes of profiles down the line).

**Comment by humamaboalraja** [comment]
any updates on that?

**Comment by dannyneira** [comment]
Hey folks, closing as a duplicate of https://github.com/warpdotdev/Warp/issues/1009

This is on our radar and being worked on so we appreciate your patience.
